onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -0,0 +1,76 @@
+
+#include "core/optimizer/graph_transformer_utils.h"
+#include "core/optimizer/identity_elimination.h"
+#include "core/optimizer/slice_elimination.h"
+#include "core/optimizer/conv_mul_fusion.h"
+#include "core/optimizer/conv_bn_fusion.h"
+#include "core/optimizer/conv_add_fusion.h"
+#include "core/optimizer/unsqueeze_elimination.h"
+
+namespace onnxruntime {
+
+namespace transformerutils {
+
+std::vector<std::unique_ptr<RewriteRule>> GenerateRewriteRules(const TransformerLevel& level, const std::vector<std::string>& custom_list) {
+  std::vector<std::unique_ptr<RewriteRule>> rules;
+
+  if (level == TransformerLevel::Optional_L1) {
+    rules.push_back(std::make_unique<EliminateIdentity>());
+    rules.push_back(std::make_unique<EliminateSlice>());
+  } else {
+    return rules;
+  }
+
+  if (!custom_list.empty()) {
+    std::vector<std::unique_ptr<RewriteRule>> filtered_list;
+    for (const auto& rule_name : custom_list) {
+      auto it = std::find_if(rules.begin(), rules.end(), [&](std::unique_ptr<RewriteRule>& item) {
+        return item->Name() == rule_name;
+      });
+      if (it != rules.end()) {
+        filtered_list.push_back(std::move(*it));
+      }
+    }
```

Comment:
Instead of doing a 'find' and adding separately, you could use std::for_each and do the push_back in the lambda.